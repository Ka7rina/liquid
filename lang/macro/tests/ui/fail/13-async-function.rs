use liquid_lang as liquid;

#[liquid::contract(version = "0.1.0")]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn constructor(&mut self) {}

        pub async fn noop(&self) {}
    }
}

fn main() {}
