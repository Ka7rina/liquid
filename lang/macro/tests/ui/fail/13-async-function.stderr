error: `async` is not supported for functions in liquid contract
  --> $DIR/13-async-function.rs:12:13
   |
12 |         pub async fn noop(&self) {}
   |             ^^^^^
